### PR TITLE
Implemented Append for Ragged<T>

### PR DESCRIPTION
**Pull request by qindazhu**
And tested Stack for Ragged<T>

### CODE REVIEW

**Comment by danpovey** [comment]
Great!!  Merging.

File: k2/csrc/ragged_ops_inl.h [hunk 1]
```diff
@@ -123,16 +150,14 @@ Ragged<T> Transpose(Ragged<T> &src) {
...
   Array1<int32_t *> src_row_splits_ptr = GetRowSplitsPtr(src_shape);
   int32_t **src_row_splits_data = src_row_splits_ptr.Data();
-  int32_t num_axes = src.NumAxes();
   Array1<T> ans_values(c, src_values.Dim());
   const T *src_data = src_values.Data();
   T *ans_data = ans_values.Data();
   auto lambda_set_values = [=] __host__ __device__(int32_t new_idx) {
-    int32_t *idx_data =
-        reinterpret_cast<int32_t *>(malloc(sizeof(int32_t) * num_axes));
+    int32_t idx_data[kMaxNumAxes];
```
Comment (qindazhu) [inline k2/csrc/ragged_ops_inl.h:157]: Replace dynamic malloc with a static array. For now I suppose max-num-axes <= 6 (actually the code just requires size 
 with NumAxes() - 1)

Comment (danpovey) [reply k2/csrc/ragged_ops_inl.h:157]: That must have slipped past review last time, I would have objected if I saw the malloc in there.
Even having every warp doing malloc would be super slow.  Every thread is a complete no-go .

Comment (qindazhu) [reply k2/csrc/ragged_ops_inl.h:157]: Yeah, that's why I update it this time. So do you think if the current approach is fine or not? Or any other suggestion?

**Comment by danpovey** [comment]
Yes, current approach is fine.

**Pull request merged**